subprojects/packagefiles/spdk-win/meson.build

```diff
@@ -0,0 +1,95 @@
+project(
+  'spdk-win',
+  'c',
+  version: '22.05',
+)
+fs = import('fs')
+
+if get_option('build_subprojects') and not fs.exists('build' / 'lib' / 'libspdk_nvme.a')
+  message('Patching ..')
+  run_command('spdk_win_patches.sh', capture: true, check: true)
+  message('Configuring and building..')
+  run_command('spdk_win_build.bat', capture: true, check: true)
+endif
```

Comment:
It seems that there is a big overlap between the spdk-win `meson.build` and the regular SPDK `meson.build`.
I think it would be great if we could just have the single spdk subproject, then you can add the few things you need on windows using if checks for: `is_windows`.
Do you think this is possible?

Reply:
@karlowich I think the issue is that we need different **revisions** of the SPDK repository. Is that correct @rs-sam ?

Reply:
@safl Oh you're right. That is a soft requirement from WPDK, that is super annoying.

Reply:
In that case, I approve of this PR